tests/v2/test_waku_rln_relay.nim

```diff
@@ -720,3 +721,41 @@ suite "Waku rln relay":
     # check that the conversion has not distorted the original value
     check:
       keypair.get().idCommitment == idCommitment
+
+  test "Read Persistent RLN credentials":
+    # create an RLN instance
+    var rlnInstance = createRLNInstance()
+    check:
+      rlnInstance.isOk == true
+
+    var key = membershipKeyGen(rlnInstance.value)
+    var empty: array[32, byte]
+    check:
+      key.isSome
+      key.get().idKey.len == 32
+      key.get().idCommitment.len == 32
+      key.get().idKey != empty
+      key.get().idCommitment != empty
+
+    debug "the generated membership key pair: ", key
+
+    var 
+      k: MembershipKeyPair
+      index: MembershipIndex
+
+    index = MembershipIndex(1)
+    k = key.get()
+
+    var rlnMembershipCredentials = RlnMembershipCredentials(membershipKeyPair: k, rlnIndex: index)
+
+    let path = "testPath.txt"
+
+    # Write RLN credentials
+    writeFile(path, pretty(%rlnMembershipCredentials))
```

Comment:
As you abstracted the readPersistentRlnCredentials, I would suggest to do the same with a writePersistentRlnCredentials: that's because in future the RlnCredential might become a more complex object to be passed directly to pretty.

Reply:
Refer ed05b6acbb44525d4ff619fd6e9d4407a8018daa

Reply:
The test still uses [writeFile](https://github.com/status-im/nwaku/blob/ed05b6acbb44525d4ff619fd6e9d4407a8018daa/tests/v2/test_waku_rln_relay.nim#L751)

Reply:
You beat me to it. Refer 5081300b3273370367fc30c3276547bede2c61fc now

Reply:
Seems to be addressed in https://github.com/status-im/nwaku/pull/1037/commits/5081300b3273370367fc30c3276547bede2c61fc